pxr/imaging/hdSt/renderBuffer.cpp

```diff
@@ -61,7 +61,9 @@ HdStRenderBuffer::HdStRenderBuffer(
 {
 }
 
-HdStRenderBuffer::~HdStRenderBuffer() = default;
+HdStRenderBuffer::~HdStRenderBuffer() {
+    _resourceRegistry->GarbageCollect();
```

Comment:
Hi!  Can you expand on the intent of this line?  Is there a benefit to invoking this here, as opposed to ~HdStRenderDelegate or something?  Thanks!

Reply:
@tcauchois Just copy pasting on Maddy's behalf

> I tested the suggested change (moving the call to _resourceRegistry->GarbageCollect(); from HdStRenderBuffer::~HdStRenderBuffer to HdStRenderDelegate::~HdStRenderDelegate). ctest -C Release still shows all tests passing, and my simple repro for the memory leak doesn't show leaking with it. 

> When debugging the leak, I put the garbage collection in HdStRenderBuffer, because that's what Instruments and stack traces were pointing me to. (HgiMetalTexture objects were being leaked because they were owned by HdStUvTextureObjects, HdStUvTextureObjects were being leaked because they were created by HdStRenderBuffer and kept alive in the HdSt_TextureObjectRegistry even after HdStRenderBuffer dropped its shared pointers to _textureObject and _textureMSAAObject in its default destructor.) To me, that points to HdStRenderBuffer being responsible for the leaked memory, including cleaning it up. But I'm not familiar with the architecture of Usd and imaging, so if people find that it makes more sense for the clean-up to be in the HdStRenderDelegate instead, that's fine.

Hopefully that makes sense, but I can ping Maddy for more.

Let me know what you think and I can adjust the PR accordingly, or however you want to handle it